### ISSUE TITLE
[a11y]: [UI Shell] [Header with Navigation] - Screen reader should announce the selection status, which is visually indicated by color

### ISSUE THREAD

**Issue by 2nikhiltom** [carbon-design-system/carbon#16885]
### Package

@carbon/react

### Browser

Firefox

### Operating System

Windows

### Package version

v1.60.3

### React version

18.02

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS

### Description

The second option in the Show/Hide Menu is visually indicated as selected, but the screen reader does not announce this selection status.
<img width="733" alt="image" src="https://github.com/carbon-design-system/carbon/assets/63502271/bbd8094c-f828-4d2c-a8fb-82e80315dfff">
Screen reader expected to indicate the selection status , which is visually presented by colour ,    by using appropriate methods , for example - "by using ARIA Current  properties 

### Reproduction/example

https://react.carbondesignsystem.com/iframe.html?viewMode=story&id=components-ui-shell-header--header-w-navigation#

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
